src/router/routes/genericResourceRoute.ts

```diff
@@ -96,10 +99,16 @@ export default class GenericResourceRoute {
                     // Get the ResourceType looks like '/Patient'
                     const resourceType = req.baseUrl.substr(1);
                     const searchParamQuery = req.query;
+
+                    const allowedResourceTypes = await this.authService.getAllowedResourceTypesForOperation({
+                        operation: 'search-type',
+                        accessToken: cleanAuthHeader(req.headers.authorization),
+                    });

```

Comment:
Do we need to add this to any bundle logic ie if you post a bundle that does a search?

Reply:
Not right now.

batch/transaction can include search queries, so passing down the allowedResourceTypes will be necessary.

However, we currently do not support batch: https://github.com/awslabs/fhir-works-on-aws-routing/blob/mainline/src/router/bundle/bundleHandler.ts#L59

And our transaction implementation does not support search: https://github.com/awslabs/fhir-works-on-aws-routing/blob/mainline/src/router/bundle/bundleParser.ts#L77